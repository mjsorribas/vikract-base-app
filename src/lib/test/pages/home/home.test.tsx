import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Home from "lib/pages/home/index";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
};

test("Header", () => {
  const component = renderer.create(
    <MemoryRouter initialEntries={[{ pathname: "/" }]}>
      <Home />
    </MemoryRouter>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
