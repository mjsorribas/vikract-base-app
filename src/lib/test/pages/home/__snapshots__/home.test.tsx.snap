// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home 1`] = `
<div
  className="css-qi60vh"
>
  <div
    className="css-164050e"
  >
    <button
      aria-label="left-arrow"
      className="chakra-button css-4n7clc"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.707 17.293 8.414 13H18v-2H8.414l4.293-4.293-1.414-1.414L4.586 12l6.707 6.707z"
        />
      </svg>
    </button>
    <button
      aria-label="right-arrow"
      className="chakra-button css-1mzvrn2"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden={true}
        fill="currentColor"
        focusable={false}
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m11.293 17.293 1.414 1.414L19.414 12l-6.707-6.707-1.414 1.414L15.586 11H6v2h9.586z"
        />
      </svg>
    </button>
    <div
      className="slick-slider slick-initialized"
      dir="ltr"
    >
      <div
        className="slick-list"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={null}
        onMouseMove={null}
        onMouseUp={[Function]}
        onTouchCancel={null}
        onTouchEnd={[Function]}
        onTouchMove={null}
        onTouchStart={[Function]}
        style={{}}
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
        >
          <div
            aria-hidden={false}
            className="slick-slide slick-active slick-current"
            data-index={0}
            onClick={[Function]}
            style={
              {
                "left": -0,
                "opacity": 1,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms ease, visibility 500ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="css-xuck0k"
                style={
                  {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              />
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={1}
            onClick={[Function]}
            style={
              {
                "left": -0,
                "opacity": 0,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms ease, visibility 500ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="css-1802qz0"
                style={
                  {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              />
            </div>
          </div>
          <div
            aria-hidden={true}
            className="slick-slide"
            data-index={2}
            onClick={[Function]}
            style={
              {
                "left": -0,
                "opacity": 0,
                "outline": "none",
                "position": "relative",
                "transition": "opacity 500ms ease, visibility 500ms ease",
                "width": 0,
              }
            }
            tabIndex="-1"
          >
            <div>
              <div
                className="css-ykn6ct"
                style={
                  {
                    "display": "inline-block",
                    "width": "100%",
                  }
                }
                tabIndex={-1}
              />
            </div>
          </div>
        </div>
      </div>
      <ul
        className="slick-dots"
        onMouseEnter={null}
        onMouseLeave={null}
        onMouseOver={null}
        style={
          {
            "display": "block",
          }
        }
      >
        <li
          className="slick-active"
        >
          <button
            onClick={[Function]}
          >
            1
          </button>
        </li>
        <li
          className=""
        >
          <button
            onClick={[Function]}
          >
            2
          </button>
        </li>
        <li
          className=""
        >
          <button
            onClick={[Function]}
          >
            3
          </button>
        </li>
      </ul>
    </div>
  </div>
  <br />
  <div
    align="start"
    className="chakra-tabs css-ci5oqm"
  >
    <div
      aria-orientation="horizontal"
      className="chakra-tabs__tablist css-fgp5ep"
      onKeyDown={[Function]}
      role="tablist"
    >
      <button
        aria-controls="tabs-:r1:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1mbanyi"
        id="tabs-:r1:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Features
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1mbanyi"
        id="tabs-:r1:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Components
      </button>
      <button
        aria-controls="tabs-:r1:--tabpanel--1"
        aria-selected={false}
        className="chakra-tabs__tab css-1mbanyi"
        id="tabs-:r1:--tab--1"
        onClick={[Function]}
        onFocus={[Function]}
        role="tab"
        tabIndex={-1}
        type="button"
      >
        Documentation
      </button>
    </div>
    <div
      className="chakra-tabs__tab-panels css-8atqhb"
    >
      <div
        aria-labelledby="tabs-:r1:--tab-0"
        className="chakra-tabs__tab-panel css-a5mhaz"
        hidden={false}
        id="tabs-:r1:--tabpanel-0"
        role="tabpanel"
        tabIndex={0}
      >
        Content 1
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-1"
        className="chakra-tabs__tab-panel css-a5mhaz"
        hidden={true}
        id="tabs-:r1:--tabpanel-1"
        role="tabpanel"
        tabIndex={0}
      >
        Content 2
      </div>
      <div
        aria-labelledby="tabs-:r1:--tab-2"
        className="chakra-tabs__tab-panel css-a5mhaz"
        hidden={true}
        id="tabs-:r1:--tabpanel-2"
        role="tabpanel"
        tabIndex={0}
      >
        Content 3
      </div>
    </div>
  </div>
  <span />
  <span
    hidden={true}
    id="__chakra_env"
  />
</div>
`;
